core/modules/wsched/ScanScheduler.cc

```diff
@@ -54,11 +54,10 @@ ChunkDisk* dbgChunkDisk1 = nullptr; ///< A symbol for gdb
 ////////////////////////////////////////////////////////////////////////
 // class ScanScheduler
 ////////////////////////////////////////////////////////////////////////
-ScanScheduler::ScanScheduler()
-    : _maxRunning(32),  // FIXME: set to some multiple of system proc count.
-      _disks(),
-      _logger(LOG_GET(getName())),
-      _mutex()
+ScanScheduler::ScanScheduler(int maxThreads)
+    : _maxThreads{maxThreads},  // FIXME: set to some multiple of system proc count.
```

Comment:
The maximum number of threads is really decided by the size of the thread pool. maxThreads is meant to be slightly smaller than the size of the thread pool so ScanScheduler doesn't completely crowd out the other schedulers ( really just GroupScheduler so far). I feel this should be set in a configuration file so it can be easily changed for different hardware and we can tweak them to see what gives best performance. Currently, all of these values are set in the SsiService constructor.